src/mca/plm/base/plm_base_jobid.c

```diff
@@ -96,13 +97,18 @@ int prrte_plm_base_create_jobid(prrte_job_t *jdata)
         found = false;
         for (i=1; i < INT16_MAX; i++) {
             ptr = NULL;
-            pjid = PRRTE_CONSTRUCT_LOCAL_JOBID(PRRTE_PROC_MY_NAME->jobid, i);
+            pjid = PRRTE_CONSTRUCT_LOCAL_JOBID(PRRTE_PROC_MY_NAME->jobid, prrte_plm_globals.next_jobid);
             prrte_hash_table_get_value_uint32(prrte_job_data, pjid, (void**)&ptr);
             if (NULL == ptr) {
-                prrte_plm_globals.next_jobid = i;
                 found = true;
                 break;
             }
+
+            if (INT16_MAX == prrte_plm_globals.next_jobid) {
```

Comment:
Somehow, this loop just looks wrong to me. I only go thru this routine if we are reusing jobids. I want to start at jobid=1 and cycle across all possible jobid's until I find the first empty one, and then set next_jobid equal to that one. Why not just do it with the index (i) and move next_jobid when complete? I guess I'm not understanding how this change helped.

Reply:
I guess I don't understand the concern about always staying at index 1 and 2. If jobs are terminating, then that is exactly what we want to do. Remember, this ultimately winds up in a hash table, so keeping the number small is a good thing unless we really have to deal with lots and lots of simultaneous jobs.

Reply:
I manually verified that it is correct. However, we can discuss if it is what we want to do or not.

Before we hit the reuse limit we increment the jobid through the range of [1, INT16_MAX-1]. This code just makes sure we do the same after the reuse limit.

A minor point maybe, but the reason I took the time to do so was for consistency and debugability above this limit. This way we have a wide range of job ids for the whole `INT16_MAX` range.

Reply:
We should probably rethink it. If we are going to cycle over the range after we hit reuse, then we might as well be doing so from the beginning. We didn't used to do so because we thought it slowed us down, but maybe that isn't true, especially if jobs are starting and ending quickly so the total number alive is low. Might be a better solution than making the hash table grow for no reason.

Worth pondering.

Reply:
If we do not cycle would the user see the same jobid for every job they submit against the same DVM instance? Since it is encoded I'd need to check.

Reply:
hmmm...that's a good point...and compelling. okay, i'm sold 😄

Reply:
I just checked. Against the same `prte` instance if we have it reuse jobids from the start then it's encoded jobid will be the same (jobid `1` is encoded the same for the two different jobs to which it was assigned). This can be confusing to users that may be using the jobid from the tagged output to differentiate between job instances.

This will break down once they hit the `INT16_MAX` limit since they will now be looping back to the beginning, but I think that is ok.